content/blog/understanding-event-loop/index.md

```diff
@@ -0,0 +1,102 @@
+
+But now let's assume that we have a function which is taking some time to execute, for example in the below function which is having a loop over 10k times, the `console.log()` would be executed once the loop is over and in the meantime, our UI interaction with the browser would be interrupted.
+
+```JavaScript
+function someTimeTakingFunc() {
+  for(var i = 0; i < 10; i++) {
+
+  }
+  console.log("Loop has been executed")
+}
+```
+
+The above scenario seems to be impractical, but there might be cases where any function can take an indeterminate amount of time and our main thread would be blocked, for example calling an API for fetching data from the server-side. And this is the practical use case we are dealing with in development every day. So how JavaScript handles this and where does the event loop comes into the picture? We will surely get to know this, but before moving further, let's understand the basic memory architecture in JavaScript.
+
+### Memory Organization of JavaScript: 
+
+The Javascript Engine consists of two main components:
+ -  **Memory Heap** — this is where the memory allocation happens, all of our object variables are assigned here in a random manner.
+ -  **Call Stack** — this is where your function calls are stored.
+
+ ### Understanding the Call Stack
+
+Call stack is a LIFO (last in first out) data structure. All of the function calls are pushed into this call stack and are said to be a frame. In short function, a stack is nothing but the simple stack data structure which keeps tracks of the function currently being executed.
+
+Let's understand the above concept with the example of the following code snippet:
+
+ ```JavaScript
+
+  function square(b) {
+    return b * b ; 
+  }
+
+  function cube(x) {
+    return x * square(x)
+  }
+
+  console.log(cube(8));
+
+ ```
+
+The above snippet is an example of defining a `cube`  function, which is calling a `square` function, to find the cube of a number passed in the argument. But how does this work in call stack? Let's understand step by step: 
+
+1. When we call the `console.log()` line, this `code/function` is pushed into the call stack. 
+ 2. Now the above `console.log()` function is called the `cube` function, hence this function is pushed into the stack. 
+ 3. After this the `cube` function is calling the `square` function, and now this would be pushed into the stack. 
+ 4. Now once the `square` function is executed, and the result is returned, then the `square` function is popped out os the stack. 
+ 5. Since at this step, we have got then `square` function result, so `cube` function would be executed and popped out of the call stack. 
+ 6. At last, the `console.log()` would be executed and this function would be popped out of the stack and the stack would now be empty.
```

Comment:
same `poppped out off` instead of `popped out of`

Reply:
I doubt this, it is either `popped out` or `popped off`. If we are using `popped out ` then of is used for prepositions not a verb